libr/socket/run.c

```diff
@@ -74,11 +74,27 @@ static void dyn_init(void) {
 		dyn_openpty = r_lib_dl_sym (NULL, "openpty");
 	}
 	if (!dyn_login_tty) {
-		dyn_openpty = r_lib_dl_sym (NULL, "login_tty");
+		dyn_login_tty = r_lib_dl_sym (NULL, "login_tty");
 	}
 	if (!dyn_forkpty) {
-		dyn_openpty = r_lib_dl_sym (NULL, "forkpty");
+		dyn_forkpty = r_lib_dl_sym (NULL, "forkpty");
 	}
+
+#ifdef __linux__
+	// for linux distros which use it, attempt to fall back on libutil.so if something failed
+	if (!(dyn_openpty && dyn_login_tty && dyn_forkpty)) {
+		r_lib_dl_open("libutil.so");
```

Comment:
check if it returns NULL; if so, you can skip the lines below

Reply:
its fine to hardcode .so here, because we are in __linux__ specific thing.. but this libutil is not just from Linux, some BSD systems also use `R_LIB_EXT` instead. and remove the ifdef __linux__ thing, because this can apply to cygwin, macOS and freebsd at least., not just linux. and afaik it all depends on how libc is compiled, so imho its better to do this thing linking at runtime